From 9a5f390a13ef483e018b47ce85b0f6a7aa9e4239 Mon Sep 17 00:00:00 2001
From: David Pacheco <dap@joyent.com>
Date: Thu, 1 Jun 2017 17:57:57 -0700
Subject: [PATCH] MANTA-3293 "manta-adm alarm config verify --unconfigure"
 crash Reviewed by: Kody A Kantor <kody.kantor@gmail.com> Reviewed by: Trent
 Mick <trentm@gmail.com> Approved by: Trent Mick <trentm@gmail.com>

---
 lib/alarms/update.js | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/lib/alarms/update.js b/lib/alarms/update.js
index 7f383b6..529aa94 100644
--- a/lib/alarms/update.js
+++ b/lib/alarms/update.js
@@ -844,16 +844,15 @@ function amonUpdatePlanSummarize(args)
 		ntotagents = 0;
 		ntotprobes = 0;
 		plan.mup_groups_remove.forEach(function (pg) {
-			assertplus.ok(plan.mup_nremove_bygroup.hasOwnProperty(
-			    pg.pg_uuid));
 			assertplus.ok(!plan.mup_nadd_bygroup.hasOwnProperty(
 			    pg.pg_uuid));
-			nprobes = plan.mup_nremove_bygroup[pg.pg_uuid];
+			nprobes = plan.mup_nremove_bygroup.hasOwnProperty(
+			    pg.pg_uuid) ?
+			    plan.mup_nremove_bygroup[pg.pg_uuid] : 0;
 
-			assertplus.ok(plan.mup_agents_bygroup.hasOwnProperty(
-			    pg.pg_uuid));
-			nagents = Object.keys(plan.mup_agents_bygroup[
-			    pg.pg_uuid]).length;
+			nagents = plan.mup_agents_bygroup.hasOwnProperty(
+			    pg.pg_uuid) ? Object.keys(
+			    plan.mup_agents_bygroup[pg.pg_uuid]).length : 0;
 
 			fprintf(out, '%7d %7d %s\n',
 			    nprobes, nagents, pg.pg_name);
-- 
2.21.0

